autoload/slime.vim

```diff
@@ -78,17 +78,28 @@ function! s:TmuxCommand(config, args)
 endfunction
 
 function! s:TmuxSend(config, text)
-  call s:WritePasteFile(a:text)
-  call s:TmuxCommand(a:config, "load-buffer " . g:slime_paste_file)
   if exists("b:slime_bracketed_paste")
     let bracketed_paste = b:slime_bracketed_paste
   elseif exists("g:slime_bracketed_paste")
     let bracketed_paste = g:slime_bracketed_paste
   else
     let bracketed_paste = 0
   endif
+
+  let last_char_is_carriage_return = 0
+  if bracketed_paste && a:text[-1:] == "\r"
+    call s:WritePasteFile(a:text[:-2])
+    let last_char_is_carriage_return = 1
+  else
+    call s:WritePasteFile(a:text)
+  end
+
+  call s:TmuxCommand(a:config, "load-buffer " . g:slime_paste_file)
   if bracketed_paste
     call s:TmuxCommand(a:config, "paste-buffer -d -p -t " . shellescape(a:config["target_pane"]))
+    if last_char_is_carriage_return
+      call s:TmuxCommand(a:config, "send-keys -t " . shellescape(a:config["target_pane"]) . " Enter")
```

Comment:
is `Enter` different from … newline?
(if the `\r` had been a `\n`, would it have worked?)

Reply:
Just added some code to also handle `\n` and `\r\n`, so `SlimeRegionSend` will work.
The Enter in tmux is an emulated key, defined here as `\r`: https://github.com/tmux/tmux/blob/3b7dae9a5339c4495d30a4c1bb28ffe8aead0b27/key-string.c#L63

Reply:
actually, if you could tell me how to reproduce both a positive and negative case — I could test this on my side too 👍

Reply:
I think they didn't act that different, only `send-keys` and `paste` acts differently. This PR  issue a `send-keys enter` if the last char is crlf instead, send-keys will not be issued as [bracketed-paste](https://en.wikipedia.org/wiki/Bracketed-paste), so works just like you pressed Enter physically. From the wiki paste mode is using special terminal keys to quote the content, it will not autoindent, faster, almost always want we want, except it just paste the content, will never execute it even it's a newline.